docs/reference/contributing/guidelines/design_guidelines.md

```diff
@@ -131,7 +131,68 @@ A general module can be split into two APIs, the frontend (or user API) and the
 
 ## Documentation
 
-- Each function and class in a module should provide a doxygen comment that documents the function and each argument and return value:
+- Document all entities in a `.h` file using doxygen comment blocks.
+
+    Doxygen comment blocks start with `/**` or `/*!` and end with `*/`, and you put them above the documented entity.
+
+    For example:
+    
+    ```
+    /** psa_crypto_ipc_s struct used for some of the
+     * PSA Crypto APIs that need psa_key_handle_t and psa_algorithm_t arguments.
+     * To use the existing infrastructure of the SPM-IPC we provide a struct to
+     * pack them together.
+     */
+    typedef struct psa_crypto_ipc_s {
+    ```    
+    
+    Each line of a long comment block usually also starts with `*`, but that is optional.
```

Comment:
We don't really have any consistency in our current docs since it is optional currently. Do want to make a stand one way or the other for all new docs going forward?

Reply:
@0xc0170 What do you think?

Reply:
@kegilbert, @AnotherButler, @0xc0170  
I have edited the text as follows to encourage consistency:
Each line of a long comment block also starts with `*`.

Reply:
Sounds good to me thanks!